cadl-extension/src/code-model-builder.ts

```diff
@@ -701,7 +701,16 @@ export class CodeModelBuilder {
   }
 
   private processParameterBody(op: CodeModelOperation, body: ModelProperty | Model, parameters: Model) {
-    const schema = this.processSchema(body.kind === "Model" ? body : body.type, body.name);
+    let schema = this.processSchema(body.kind === "Model" ? body : body.type, body.name);
+    // handle for binary body
+    if (
+      parameters.properties.has("contentType") &&
+      body.kind === "ModelProperty" &&
+      body.type.kind === "Scalar" &&
+      body.type.name === "bytes"
+    ) {
+      schema = new BinarySchema(body.name);
+    }
```

Comment:
1. should be an `if else` for the schema
2. do we need to test "contentType", probably just drop it (and even if we need to test, "contentType" is not a wire name; also then we would need to test again "application/json" or anything relates to JSON, as it is totally valid for user to write `content-type: application/json` even it is not necessary)?

Reply:
If user uses `application/json` and implicit body (as we most use here), the body would be `kind=Model`

(want to make sure we still can distinguish `@body data: bytes`, from `data: bytes` in an implicit Model for JSON)

Reply:
Use `if else`. Don't process schema you do not use.